lms/services/http.py

```diff
@@ -31,62 +55,6 @@ def patch(self, *args, **kwargs):
     def delete(self, *args, **kwargs):
         return self.request("DELETE", *args, **kwargs)
 
-    def request(self, method, url, timeout=(10, 10), **kwargs):
-        r"""
-        Send a request with `requests` and return the requests.Response object.
-
-        :param method: The HTTP method to use, one of "GET", "PUT", "POST",
-            "PATCH", "DELETE", "OPTIONS" or "HEAD"
-        :param url: The URL to request
-        :param timeout: How long (in seconds) to wait before raising an error.
-
-            This can be a (connect_timeout, read_timeout) 2-tuple or it can be
-            a single float that will be used as both the connect and read
-            timeout.
-
-            Good practice is to set this to slightly larger than a multiple of
-            3, which is the default TCP packet retransmission window. See:
-            https://docs.python-requests.org/en/master/user/advanced/#timeouts
-
-            Note that the read_timeout is *not* a time limit on the entire
-            response download. It's a time limit on how long to wait *between
-            bytes from the server*. The entire download can take much longer.
-        :param \**kwargs: Any other keyword arguments will be passed directly to
-            requests.Session().request():
-            https://docs.python-requests.org/en/latest/api/#requests.Session.request
-        :raise ExternalRequestError: If sending the request or receiving the
-            response fails (DNS failure, refused connection, timeout, too many
-            redirects, etc).
-
-            The original exception from requests will be available as
-            ExternalRequestError.__cause__.
-
-            In this case ExternalRequestError.response will be None.
-
-        :raise ExternalRequestError: If an error response (4xx or 5xx) is
-            received.
-
-            The original exception from requests will be available as
-            ExternalRequestError.__cause__.
-
-            The error response will be available as
-            ExternalRequestError.response.
```

Comment:
I found this doc-string too long to be useful. There's some interesting and useful info here, but I don't think this is the right place for a tutorial.

The detail in the `ExternalRequestError` seems to be going into things which are true, but I'm not sure why you need to read about them here. There are many other true statements we could include here about how all these objects work, but it's 8 lines of code, so perhaps it's not proportionate?